.github/workflows/deploy_tre_reusable.yml

```diff
@@ -129,6 +132,80 @@ jobs:
+            echo "Missing secret: AAD_TENANT_ID" && exit 1
+          fi
+          if [ "${{ secrets.ACR_NAME }}" == '' ]; then
+            echo "Missing secret: ACR_NAME" && exit 1
+          fi
+          if [ "${{ secrets.DEVCONTAINER_TAG }}" == '' ]; then
+            echo "Missing secret: DEVCONTAINER_TAG" && exit 1
+          fi
+          if [ "${{ secrets.API_CLIENT_ID }}" == '' ]; then
+            echo "Missing secret: API_CLIENT_ID" && exit 1
+          fi
+          if [ "${{ secrets.API_CLIENT_SECRET }}" == '' ]; then
+            echo "Missing secret: API_CLIENT_SECRET" && exit 1
+          fi
+          if [ "${{ secrets.APPLICATION_ADMIN_CLIENT_ID }}" == '' ]; then
+            echo "Missing secret: APPLICATION_ADMIN_CLIENT_ID" && exit 1
+          fi
+          if [ "${{ secrets.APPLICATION_ADMIN_CLIENT_SECRET }}" == '' ]; then
+            echo "Missing secret: APPLICATION_ADMIN_CLIENT_SECRET" && exit 1
+          fi
+          if [ "${{ secrets.CORE_ADDRESS_SPACE }}" == '' ]; then
+            echo "Missing secret: CORE_ADDRESS_SPACE" && exit 1
+          fi
+          if [ "${{ secrets.LOCATION }}" == '' ]; then
+            echo "Missing secret: LOCATION" && exit 1
+          fi
+          if [ "${{ secrets.MGMT_RESOURCE_GROUP_NAME }}" == '' ]; then
+            echo "Missing secret: MGMT_RESOURCE_GROUP_NAME" && exit 1
+          fi
+          if [ "${{ secrets.MGMT_STORAGE_ACCOUNT_NAME }}" == '' ]; then
+            echo "Missing secret: MGMT_STORAGE_ACCOUNT_NAME" && exit 1
+          fi
+          if [ "${{ secrets.SWAGGER_UI_CLIENT_ID }}" == '' ]; then
+            echo "Missing secret: SWAGGER_UI_CLIENT_ID" && exit 1
+          fi
+          if [ "${{ secrets.TEST_APP_ID }}" == '' ]; then
+            echo "Missing secret: TEST_APP_ID" && exit 1
+          fi
+          if [ "${{ secrets.TEST_WORKSPACE_APP_ID }}" == '' ]; then
+            echo "Missing secret: TEST_WORKSPACE_APP_ID" && exit 1
+          fi
+          if [ "${{ secrets.TEST_WORKSPACE_APP_SECRET }}" == '' ]; then
+            echo "Missing secret: TEST_WORKSPACE_APP_SECRET" && exit 1
+          fi
+          if [ "${{ secrets.TEST_ACCOUNT_CLIENT_ID }}" == '' ]; then
+            echo "Missing secret: TEST_ACCOUNT_CLIENT_ID" && exit 1
+          fi
+          if [ "${{ secrets.TEST_ACCOUNT_CLIENT_SECRET }}" == '' ]; then
+            echo "Missing secret: TEST_ACCOUNT_CLIENT_SECRET" && exit 1
+          fi
+          if [ "${{ secrets.TRE_ADDRESS_SPACE }}" == '' ]; then
+            echo "Missing secret: TRE_ADDRESS_SPACE" && exit 1
+          fi
+          if [ "${{ secrets.TRE_ID }}" == '' ]; then
+            echo "Missing secret: TRE_ID" && exit 1
+          fi
+          if [ "${{ secrets.AZURE_CREDENTIALS }}" == '' ]; then
+            echo "Missing secret: AZURE_CREDENTIALS" && exit 1
+          fi
+
+      - name: Report check status start
+        if: inputs.prHeadSha != ''
+        uses: LouisBrunner/checks-action@v1.5.0
```

Comment:
If this action isn't already used at the very end of the workflow, then we don't need it here also.